docs/DeveloperGuide.md

```diff
@@ -303,179 +303,292 @@ Priorities: High (must have) - `* * * *`, Medium (nice to have) - `* * *`, Mediu
+
+**MSS**
+1. User chooses to undo the previous change.
+2. User enters the request to undo the previous change.
+3. Harmonia restores the state before the previous change.
 
     Use case ends.
 
 **Extensions**
+* 2a. Harmonia is unable to recognize the request entered by the user.
+    * 2a1. Harmonia outputs an error message.
+    * 2a2. User enters a new request.
+  Steps 2a1-2a2 are repeated until valid request is inputted.
+  Use case resumes from step 3.
+
+<br/>
+
+**Use case: UC10 - Search for tasks within date range**
+
+**MSS**
+1. User chooses to search for tasks that fall within a specific date range.
+2. User enters the request to search for the tasks by date range.
+3. Harmonia displays the tasks that match the criteria.
+4. Harmonia informs the user that the result has been displayed successfully.
 
+    Use case ends.
+
+**Extensions**
 * 2a. Harmonia detects an error in the entered request.
     * 2a1. Harmonia outputs an error message.
-    * 2a2. User enters new data.
+    * 2a2. User enters a new request.
+  Steps 2a1-2a2 are repeated until valid request is inputted.
+  Use case resumes from step 3.
 
-  Steps 2a1-2a2 are repeated until valid data is inputted.
+<br/>
+
+**Use case: UC11 - Sort tasks**
 
+**MSS**
+1. User chooses to sort the task by a certain property.
+2. User enters the request to sort the task based on a property.
+3. Harmonia displays the tasks based on the property.
+4. Harmonia informs the user that the sort has been performed successfully.
+
+    Use case ends.
+
+**Extensions**
+* 2a. Harmonia detects an error in the entered request.
+    * 2a1. Harmonia outputs an error message.
+    * 2a2. User enters a new request.
+  Steps 2a1-2a2 are repeated until valid request is inputted.
   Use case resumes from step 3.
 
-**Use case: UC08 - Searching for a task by keyword/tag**
+<br/>
 
-**MSS**
+**Use case: UC12 - Set repeated occurrences of a task**
 
-1.  User enters the request to search for task(s) by keyword/tag.
-2.  Harmonia shows a list of tasks that match the specified keyword/tag.
-3.  Harmonia informs the user that the search result has been displayed successfully.
+**MSS**
+1. User chooses the set repeated occurrence of a task. 
+2. User enters the request to set repeated occurrences of a task. 
+3. Harmonia sets the task as a repeated occurrence. 
+4. Harmonia informs the user that the task has been added as a repeated occurrence.
 
     Use case ends.
 
 **Extensions**
+* 2a. Harmonia detects an error in the entered request.
+    * 2a1. Harmonia outputs an error message.
+    * 2a2. User enters a new request.
+      Steps 2a1-2a2 are repeated until valid request is inputted.
+      Use case resumes from step 3.
+
+<br/>
+
+**Use case: UC13 - Categorise task according to priority**
+
+**MSS**
+1. User chooses to assign a priority to a task.
+2. Harmonia enters the requests to assign priority to the task.
+3. Harmonia informs the priority of the task has been updated.
```

Comment:
Not very sure about this, but will this be more consistent with the other use cases? 😅 

1. User chooses to assign a priority to a task.
2. User enters the request to assign priority to the task.
3. Harmonia assigns priority to the task.
4. Harmonia informs the user that the priority of the task has been updated.

Reply:
Fixed all of them! Thanks for your input! :-)